CI failure on double-run (tdat)

This may not happen every time, but the double run failed for #18023 in a way that seemed clearly unrelated to the PR -- see https://github.com/astropy/astropy/actions/runs/14539129250/job/40793495095

It may have something to do with uncleared state, not sure, just reporting for now.

Excerpt of the error:
```
________________________________ test_read_tdat ________________________________
[gw2] linux -- Python 3.11.12 /home/runner/work/astropy/astropy/.tox/py311-test-double/bin/python3

    def test_read_tdat():
        """Ensure the Table is as it should be
        TDAT -> Table
        """
>       assert test_table.meta["keywords"] == OrderedDict(
            [
                ("table_name", "heasarc_simple"),
                ("table_description", "Test table"),
                ("table_security", "public"),
                ("parameter_defaults", "name ra dec"),
                ("frequency_regime", "Gamma-ray"),
                ("observatory_name", "GAMMA-RAY BURSTS"),
                ("row_type", "GRB"),
                ("table_author", "Example et al."),
                ("table_priority", "3.01"),
                ("table_type", "Observation"),
                ("unique_key", "record_number"),
            ]
        )
E       AssertionError: assert {'field_delim... ra dec', ...} == OrderedDict([...ord_number')])
E         
E         Omitting 11 identical items, use -vv to show
E         Left contains 1 more item:
E         {'field_delimiter': ','}
E         
E         Full diff:
E         - OrderedDict({
E         + {
E         +     'field_delimiter': ',',
E               'frequency_regime': 'Gamma-ray',
E               'observatory_name': 'GAMMA-RAY BURSTS',
E               'parameter_defaults': 'name ra dec',
E               'row_type': 'GRB',
E               'table_author': 'Example et al.',
E               'table_description': 'Test table',
E               'table_name': 'heasarc_simple',
E               'table_priority': '3.01',
E               'table_security': 'public',
E               'table_type': 'Observation',
E               'unique_key': 'record_number',
E         - })
E         + }

../../.tox/py311-test-double/lib/python3.11/site-packages/astropy/io/ascii/tests/test_tdat.py:166: AssertionError
```

--- a/astropy/io/ascii/tests/test_tdat.py
+++ b/astropy/io/ascii/tests/test_tdat.py
@@ -417,12 +417,13 @@ def test_bad_delimiter():
     ):
         test_table.write(out, format="ascii.tdat", delimiter=",")
 
-    test_table.meta["keywords"]["field_delimiter"] = ","
+    check_table = test_table.copy()
+    check_table.meta["keywords"]["field_delimiter"] = ","
     with pytest.warns(
         TdatFormatWarning, match="Delimiters other than the pipe character"
     ):
         with pytest.warns(TdatFormatWarning, match="Skipping deprecated"):
-            test_table.write(out, format="ascii.tdat", delimiter=",")
+            check_table.write(out, format="ascii.tdat", delimiter=",")
 
 
 def test_bad_header_start():

